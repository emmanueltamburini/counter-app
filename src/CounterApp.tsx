import { MouseEvent, useState } from "react";

interface Props {
    value: number;
}

export const CounterApp = ({value}: Props) => {
  const [counter, setCounter] = useState<number>(value);
  const handleAdd = (event: MouseEvent<HTMLButtonElement, globalThis.MouseEvent>) => {
    console.log(event);
    setCounter(c => c + 1);
  };

  const handleSubtract = (event: MouseEvent<HTMLButtonElement, globalThis.MouseEvent>) => {
    console.log(event);
    setCounter(c => c -1);
  };

  const handleReset = (event: MouseEvent<HTMLButtonElement, globalThis.MouseEvent>) => {
    console.log(event);
    setCounter(value);
  };

  return (
    <>
        <h1>CounterApp</h1>
        <h2> { counter } </h2>
        <button aria-label="btn-subtract" onClick={handleSubtract}>
            -1
        </button>
        <button aria-label="btn-add" onClick={handleAdd}>
            +1
        </button>
        <button aria-label="btn-reset" onClick={handleReset}>
            reset
        </button>
    </>
  )
}
