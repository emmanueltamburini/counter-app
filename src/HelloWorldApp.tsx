const newMessage = {
    message: "Hello World",
    title: "Fernando"
};

const helloWorld = () => "Hello world";

const HelloWorldApp = () => {
  return (
    <>
        <h1>{helloWorld()}</h1>
        <code>{JSON.stringify(newMessage)}</code>
        <p>I am a subtitle</p>
    </>
  )
}

export default HelloWorldApp;