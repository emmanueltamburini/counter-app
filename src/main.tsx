import React from 'react';
import ReactDOM  from 'react-dom/client';
import { CounterApp } from './CounterApp';
import "./styles.css";
import HelloWorldApp from './HelloWorldApp';

const element = document.getElementById('root');

if (element) {
    ReactDOM.createRoot(element).render(
        <React.StrictMode>
            <HelloWorldApp title='Title' subTitle={1} name='Emmanuel'/>
        </React.StrictMode>
    )
}
