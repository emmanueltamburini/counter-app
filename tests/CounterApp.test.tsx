import React from "react";
import { fireEvent, render, screen } from "@testing-library/react"
import { CounterApp } from '../src/CounterApp';

describe('CounterApp test', () => {
    const initialValue = 100;
    test('should be equal to snapshot', () => {
        const {container} = render(<CounterApp value={initialValue}/>);
        expect(container).toMatchSnapshot();
    });

    test('should show the value "initialValue"', () => {
        render(<CounterApp value={initialValue}/>);
        expect(screen.getByText(initialValue)).toBeTruthy();
        expect(screen.getByRole('heading', {level: 2}).innerHTML).toContain(initialValue.toString());
    });

    test('should increment with +1 button', () => {
        render(<CounterApp value={initialValue}/>);
        fireEvent.click(screen.getByText('+1'));
        expect(screen.getByText(initialValue + 1)).toBeTruthy();
    });

    test('should decrement with +1 button', () => {
        render(<CounterApp value={initialValue}/>);
        fireEvent.click(screen.getByText('-1'));
        expect(screen.getByText(initialValue - 1)).toBeTruthy();
    });

    test('should reset button works', () => {
        render(<CounterApp value={initialValue}/>);
        fireEvent.click(screen.getByRole('button', {name: 'btn-add'}));
        fireEvent.click(screen.getByRole('button', {name: 'btn-add'}));
        fireEvent.click(screen.getByRole('button', {name: 'btn-subtract'}));
        expect(screen.getByText(initialValue + 1)).toBeTruthy();
        fireEvent.click(screen.getByRole('button', {name: 'btn-reset'}));
        expect(screen.getByText(initialValue)).toBeTruthy();
    });
})