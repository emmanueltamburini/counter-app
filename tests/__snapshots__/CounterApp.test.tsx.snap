// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterApp test should be equal to snapshot 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2>
     
    100
     
  </h2>
  <button
    aria-label="btn-subtract"
  >
    -1
  </button>
  <button
    aria-label="btn-add"
  >
    +1
  </button>
  <button
    aria-label="btn-reset"
  >
    reset
  </button>
</div>
`;
